packages/iris-grid/src/IrisGridPartitionedTableModel.ts

```diff
@@ -0,0 +1,442 @@
+    return this.model.selectDistinctColumns;
+  }
+
+  set selectDistinctColumns(names: readonly ColumnName[]) {
+    this.model.selectDistinctColumns = names;
+  }
+
+  get pendingDataMap(): PendingDataMap {
+    return this.model.pendingDataMap;
+  }
+
+  set pendingDataMap(map: PendingDataMap) {
+    this.model.pendingDataMap = map;
+  }
+
+  get pendingRowCount(): number {
+    return this.model.pendingRowCount;
+  }
+
+  set pendingRowCount(count: number) {
+    this.model.pendingRowCount = count;
+  }
+
+  get pendingDataErrors(): PendingDataErrorMap {
+    return this.model.pendingDataErrors;
+  }
+
+  commitPending(): Promise<void> {
+    return this.model.commitPending();
+  }
+
+  isFilterable(columnIndex: ModelIndex): boolean {
+    return this.model.isFilterable(columnIndex);
+  }
+
+  setViewport(
+    top: VisibleIndex,
+    bottom: VisibleIndex,
+    columns?: Column[]
+  ): void {
+    this.model.setViewport(top, bottom, columns);
+  }
+
+  snapshot(ranges: readonly GridRange[]): Promise<readonly unknown[][]> {
+    return this.model.snapshot(ranges);
+  }
+
+  textSnapshot(
+    ranges: readonly GridRange[],
+    includeHeaders?: boolean,
+    formatValue?: (value: unknown, column: Column, row?: Row) => string
+  ): Promise<string> {
+    return this.model.textSnapshot(ranges, includeHeaders, formatValue);
+  }
+
+  valuesTable(columns: Column | readonly Column[]): Promise<Table> {
+    return this.model.valuesTable(columns);
+  }
+
+  close(): void {
+    this.partitionedTable.close();
+  }
+
+  isRowMovable(): boolean {
+    return this.model.isRowMovable();
+  }
+
+  delete(ranges: readonly GridRange[]): Promise<void> {
+    return this.model.delete(ranges);
+  }
+
+  seekRow(
+    startRow: number,
+    column: Column,
+    valueType: ValueTypeUnion,
+    value: unknown,
+    insensitive?: boolean,
+    contains?: boolean,
+    isBackwards?: boolean
+  ): Promise<number> {
+    return this.model.seekRow(
+      startRow,
+      column,
+      valueType,
+      value,
+      insensitive,
+      contains,
+      isBackwards
+    );
+  }
+
+  get isSeekRowAvailable(): boolean {
+    return this.model.isSeekRowAvailable;
+  }
+
+  get columnHeaderGroups(): readonly ColumnHeaderGroup[] {
+    return this.model.columnHeaderGroups;
+  }
+
+  set columnHeaderGroups(groups: readonly ColumnHeaderGroup[]) {
+    this.model.columnHeaderGroups = groups;
+  }
+
+  get columnHeaderGroupMap(): ReadonlyMap<string, ColumnHeaderGroup> {
+    return this.model.columnHeaderGroupMap;
+  }
+
+  getColumnHeaderParentGroup(
+    modelIndex: ModelIndex,
+    depth: number
+  ): ColumnHeaderGroup | undefined {
+    return this.model.getColumnHeaderParentGroup(modelIndex, depth);
+  }
+
+  dataBarOptionsForCell(
+    column: number,
+    row: number,
+    theme: IrisGridThemeType
+  ): DataBarOptions {
+    return this.model.dataBarOptionsForCell(column, row, theme);
+  }
+
+  get partitionColumns(): Column[] {
```

Comment:
Enforce returning a `readonly` array
```suggestion
  get partitionColumns(): readonly Column[] {
```